main.go

```diff
@@ -48,13 +52,15 @@ func verify(ctx context.Context,
 	}
 
 	// Verify the workflow identity.
-	if err := pkg.VerifyWorkflowIdentity(workflowInfo, source); err != nil {
+	builderID, err := pkg.VerifyWorkflowIdentity(workflowInfo, builderOpts,
```

Comment:
i think we will need to revisit this eventually and generalize this to "VerifySignerIdentity" so that it can both check github workflow signed certificates against the source/builder, but also support general other builders that sign with other types of certifiates and keys. wdyt?

Reply:
Agreed. The function is doing 2 things at once (builder and source). I did not want to change it in this PR. Ill try to do tat with the GCB support PR